### DISCUSSION TITLE
Custom rules for test sourceSet.

### DISCUSSION THREAD

**Discussion by MaximSysoev** [detekt/detekt#6930] [General]
Hi everyone! 

I have a question about custom rules. I have an android project. I have created my custom rule set. And I want it to run on an androidTest source set.
I have included my module to my detekt convension plugin by this line:
`dependencies.add("detektPlugins", dependencies.project(plugin))`
And configure task with these source sets:
```
"src/main/kotlin",
"src/test/kotlin"
```
I want to run regular detekt rules on these source sets, but my custom rules on `src/androidTest/kotlin`.
How can I do this?

**Comment by cortinico** [comment]
> I want to run regular detekt rules on these source sets, but my custom rules on src/androidTest/kotlin.

Do you want to run with Type Resolution or without, as the answer is different

**Reply by MaximSysoev** [reply]
Without type resolution.

**Reply by cortinico** [reply]
> And configure task with these source sets:

Could you share how you configured the task?

**Reply by MaximSysoev** [reply]
About this:
```kotlin
private fun Project.configureTasks() {
    val htmlReportFile = layout.buildDirectory.file("detekt-results.html")
    val openReportTask = configureOpenReportTask(htmlReportFile)
    tasks.withType(Detekt::class.java) { detekt ->
      detekt.jvmTarget = getTargetJvmVersion().toString()
      detekt.excludeSources()
      detekt.setSources(this)
      detekt.setupHtmlReport(htmlReportFile)
      detekt.finalizedBy(openReportTask)
    }
    tasks.withType(Jar::class.java) { jar ->
      jar.duplicatesStrategy = DuplicatesStrategy.EXCLUDE
    }
    configureDetektTask()
  }
  
  private fun Project.configureDetektTask() {
    val taskNames = listOf("detektTest", "detektDebug", "detektGoogleDebug", "detekt")

    tasks.register(RunDetektTaskName) { detektTask ->
      detektTask.dependsOn(
        tasks.withType(Detekt::class.java).matching { it.name in taskNames },
      )
      detektTask.dependsOn(":custom-detekt-rules:assemble")
    }
  }

  private fun Detekt.setSources(project: Project) {
    setSource(
      project.files(
        "src/main/kotlin",
        "src/test/kotlin",
      ),
    )
  }
```

**Reply by cortinico** [reply]
Have you tried following the documentation here?
https://detekt.dev/docs/gettingstarted/gradle#leveraging-gradles-sourcetask---excluding-and-including-source-files

**Reply by MaximSysoev** [reply]
Yes, I saw it, thanks. I will try to separate my configuration on two configurations for detektTest and others.

**Reply by MaximSysoev** [reply]
It seems that I understood detektTest wrong :D
It works only for type resolution, now I see. So I should use just `includes: "**/androidTest/**"` for my test rules, right?
